### ISSUE TITLE
v15.8: Debugger crashes with BoundsError

### ISSUE THREAD

**Issue by zhopan77** [julia-vscode/julia-vscode#1072]
Versions:
* Windows 10 64-bit
* Julia 1.4
* VSCode 1.43.1
* VSCode Julia Insider 0.15.8

The debugger of Julia Insider 0.15.6 were working fine. 

Automatic upgrade to 0.15.7 happened when it became available, but Julia language server would crash multiple times within a minute.

So I manually upgraded to 0.15.8 which seemed to work for other functionalities (e.g. hover help, go to definition, etc.). However, the visual debugger would crash at random locations in the source code when I stepped through the code by pushing F10 and sometimes F11. The following error message was shown. This didn't happen for 0.15.6.

`ERROR: BoundsError: attempt to access 0-element Array{Main.VSCodeDebugger.VariableReference,1} at index [1]
Stacktrace:
 [1] getindex at .\array.jl:787 [inlined]
 [2] getvariables_request(::Base.PipeEndpoint, ::Main.VSCodeDebugger.DebuggerState, ::String, ::String) at c:\Users\zpan\.vscode\extensions\julialang.language-julia-insider-0.15.18\scripts\debugger\debugger_requests.jl:365
 [3] startdebug(::String) at c:\Users\zpan\.vscode\extensions\julialang.language-julia-insider-0.15.18\scripts\debugger\debugger.jl:80
 [4] top-level scope at c:\Users\zpan\.vscode\extensions\julialang.language-julia-insider-0.15.18\scripts\debugger\run_debugger.jl:6
 [5] include(::Module, ::String) at .\Base.jl:377
 [6] exec_options(::Base.JLOptions) at .\client.jl:288
 [7] _start() at .\client.jl:484

Julia debuggee finished. Press ENTER to close this terminal.`

**Comment by zhopan77** [comment]
One more detail: I enabled compile mode during the debugging.

**Comment by phlavenk** [comment]
I'm seeing the reported crash regularly. Same configuration, 0.15.19

**Comment by phlavenk** [comment]
Also with compiled-mode = on

**Comment by davidanthoff** [comment]
I'm trying to reproduce this, not luck so far.

Could one of you try to come up with a simple example of how to trigger this error? Ideally a one file example that you could share here, with exact instructions which steps I should take to make things crash. That would be awesome!

**Comment by phlavenk** [comment]
This is a very hard-to-reproduce bug. I've seen this behavior often enough that it's annoying but not that often I can fully reproduce. And happened mostly on a larger-scale (private) project. Still don't know how to reproduce it and if it relates to some longer term history of repl/vscode/debugger. I'll comment more when I get more information. 

Am I right with my assumption, that, as debugger is loaded to the "user" julia session, this crash is not reported to the telemetry?

**Comment by davidanthoff** [comment]
Yes, you are correct, we don't have crash reporting hooked up for the debugger yet. I was actually thinking a bit about that yesterday, I might give it a shot soon.

**Comment by janbruedigam** [comment]
I also experience this error from time to time and to me it seems like this happens when you step over a line to quickly while the line is still being evaluated. But it might also be something completely different.
I have some dummy code with which I can somewhat reliably reproduce the error, but it does not always fail: 
```julia
function l1(coll)
    for el in coll
        el*el
    end
    return coll
end

function l2(coll)
    for el in coll
        l1(el)
    end
    return coll
end

function l3()
    coll = [[rand(50,50) for i=1:10] for j=1:10]
    for el in coll
        l2(el)
    end
    return
end
```

I start Julia, create the functions, step into `l3()` with `@enter l3()` and then press and hold down F10 the entire time until it breaks (or finishes the function correctly). Changing the size of the `rand(50,50)` matrix might help with reproducing the error since that makes the execution faster or slower. 
I can also reproduce the error inside a function of a package by pressing and holding down F10, but it also does not break every time.

Screenshot of the dummy code error:
![error_msg3](https://user-images.githubusercontent.com/46260565/83905569-2d7cab00-a762-11ea-9388-9cb9d34f16fb.png)

Screenshot of the package function error:
![error_msg2](https://user-images.githubusercontent.com/46260565/83905662-5b61ef80-a762-11ea-9143-2496f59788c9.png)

**Comment by xgdgsc** [comment]
I have this with an unbelievably simple code:
```
@enter begin
    for a in 1:100
        b = a + 1
    end
end
```

**Comment by xgdgsc** [comment]
It seems if I press F10 slowly it's fine. When I keep  pressing F10 it would crash entire julia REPL with no actual log shown.

**Comment by davidanthoff** [comment]
Closing as stale: this was reported against extension 0.15.8 on Julia 1.4, and there has been no activity here in several years. We cannot verify it against the current extension.

Please reopen with a current reproducer if this still happens.
